Makefile.rules

```diff
@@ -125,65 +125,45 @@ dosetrev:
 
 .PHONY: docbook-clean
 docbook-clean:
-	-@for r in $(modules) $(static_modules_path) "" ; do \
-		if [ -d "$$r" ]; then \
-			if [ -d "$$r"/doc ]; then \
-				rm -f "$$r"/doc/*.txt ; \
-				rm -f "$$r"/doc/*.html ; \
-				rm -f "$$r"/doc/*.pdf ; \
-			fi ; \
-		fi ; \
+	-@for r in $(modules) $(static_modules_path); do \
+		rm -f "$$r"/doc/*.{txt,html,pdf}; \
 	done
 
 .PHONY: dbschema-docbook-clean
 dbschema-docbook-clean:
 	-@if [ -d doc/database ] ; then \
-		rm -f doc/database/*.txt ; \
-		rm -f doc/database/*.html ; \
-		rm -f doc/database/*.pdf ; \
+		rm -f doc/database/*.{txt,html,pdf}; \
 		$(MAKE) -C db/schema docbook_clean; \
 	fi
 
+.PHONY: bin-clean
+bin-clean:
+	rm -f $(NAME)
+
 .PHONY: clean
-clean: docbook-clean dbschema-docbook-clean
-	-@rm -f $(objs) $(NAME) $(objs:.o=.il) 2>/dev/null
-	-@for r in $(all_modules) "" ; do \
-		if [ -d "$$r" -a -f "$$r/Makefile" ]; then \
-			$(MAKE) -C $$r clean ; \
-		fi ; \
-	done
+clean: docbook-clean dbschema-docbook-clean bin-clean
+	find . -type f \( -name "*.o" -o -name "*.so" \) -delete
```

Comment:
I wouldn't go with a find here - it might delete things that you don't actually want to.